algorithms/kernel/lasso_regression/lasso_regression_train_dense_default_impl.i

```diff
@@ -151,17 +146,18 @@ services::Status TrainBatchKernel<algorithmFPType, method, cpu>::compute(
 
         TlsMem<algorithmFPType, cpu, services::internal::ScalableCalloc<algorithmFPType, cpu> > tlsData(nFeatures);
         daal::threader_for(nBlocks, nBlocks, [&](const size_t iBlock) {
-            algorithmFPType * sum  = tlsData.local();
-            const size_t startRow  = iBlock * blockSize;
-            const size_t finishRow = (iBlock + 1 == nBlocks ? nRows : (iBlock + 1) * blockSize);
-            for (size_t i = startRow; i < finishRow; i++)
+            algorithmFPType * sum   = tlsData.local();
+            const size_t startRow   = iBlock * blockSize;
+            const size_t finishRow  = (iBlock + 1 == nBlocks ? nRows : (iBlock + 1) * blockSize);
+            DAAL_INT numRowsInBlock = finishRow - startRow;
+
+            daal::internal::ReadRows<algorithmFPType, cpu> xBD(xTrain.get(), startRow, numRowsInBlock);
+            DAAL_CHECK_BLOCK_STATUS_THR(xBD);
+            const algorithmFPType * xPtr = xBD.get();
+
+            for (size_t j = 0; j < nFeatures; j++)
             {
-                PRAGMA_IVDEP
-                PRAGMA_VECTOR_ALWAYS
-                for (size_t j = 0; j < nFeatures; j++)
-                {
-                    sum[j] += xPtr[i * nFeatures + j];
-                }
+                sum[j] += daal::internal::Blas<algorithmFPType, cpu>::xxdot(&numRowsInBlock, xPtr + j, &p, &ione, &zero);
```

Comment:
How much perf gain we obtain by usage `xxdot` here with parameters `&ione, &zero` instead off simple summation?